netty/src/main/java/io/grpc/netty/NettyClientTransport.java

```diff
@@ -97,6 +101,11 @@
     this.maxHeaderListSize = maxHeaderListSize;
     this.authority = new AsciiString(authority);
     this.userAgent = new AsciiString(GrpcUtil.getGrpcUserAgent("netty", userAgent));
+    if (proxyAddress != null) {
+      this.proxyHandler = new HttpProxyHandler(proxyAddress, proxyUsername, proxyPassword);
```

Comment:
The destination address isn't specified. It looks like you have to call `proxyHandler.connect(...)` after it is added to the channel. It also looks like HttpProxyHandler [always uses the same name](https://github.com/netty/netty/blob/4.1/handler-proxy/src/main/java/io/netty/handler/proxy/HttpProxyHandler.java#L123) for the CONNECT path and the Host header. Depending on how the OkHttp discussion goes, we may need to enhance the implementation or make a copy.